src/DistributedLock.Redis/RedisDistributedSemaphore.cs

```diff
@@ -39,6 +39,19 @@ public RedisDistributedSemaphore(RedisKey key, int maxCount, IDatabase database,
     /// </summary>
     public string Name => this.Key.ToString();
 
+    public int GetCurrentCount()
+    {
+        return GetCurrentCountAsync().GetAwaiter().GetResult();
+    }
```

Comment:
Should be:

```
public int GetCurrentCount() => SyncViaAsync.Run(s => s.GetCurrentCountAsync(), state: this);
```

This avoids the sync _over_ async pattern that GetResult() is, which is inefficient and can result in thread starvation.